experiment/simulation/js/main.js

```diff
@@ -1 +1,522 @@
+    return;
+  }
+  iter2++;
+  document.getElementById("instruction").innerHTML = observationMessages[iter2];
+  document.getElementById("observation").innerHTML = observationMessages[iter2];
+}
+
+function screenWidth() {
+  divWidth = document.getElementById("workspace").clientWidth;
+}
+
+let originalSimulationHeight =
+  document.getElementById("simulation").clientHeight;
+
+let restartAnimation = false;
+
+async function restart() {
+  document.getElementById("simulation").style.height = originalSimulationHeight;
+  document.getElementById("animation-video").style.display = "none";
+  document.getElementById("plotted-graph-window").style.display = "none";
+
+  document.getElementById("head-instructions").innerHTML = "Instructions";
+  document.getElementById("head-observations").innerHTML = "Instructions";
+  document.getElementById("instruction").innerHTML = "";
+  document.getElementById("observation").innerHTML = "";
+  overallIteration = -3;
+  iter2 = -1;
+  iter1 = -1;
+  setup = 0;
+  setupMessage();
+  document.getElementById("syringe").style.opacity = 0;
+  document.getElementById("apparatus-bottles").style.display = "block";
+  document.getElementById("apparatus-spectrometer").style.display = "block";
+  document.getElementById("sample-beaker").style.visibility = "hidden";
+  document.getElementById("solvent-beaker").style.visibility = "hidden";
+  document.getElementById("solution-beaker").style.visibility = "hidden";
+  document.getElementById("spectrometer").style.visibility = "hidden";
+  restartAnimation = true;
+
+  document.getElementById("solvent-beaker").style.cursor = "pointer";
+  document.getElementById("sample-beaker").style.cursor = "default";
+  document.getElementById("solution-beaker").style.cursor = "default";
+
+  // Resetting the Solution Beaker
+  document.getElementById("pink-bottom").style.fill = "none";
+  document.getElementById("layer-above-pink").style.fill = "none";
+}
+
+async function observe() {
+  if (overallIteration === 7) {
+    document.getElementById("slidecontainer").style.display = "block";
+    document.getElementById("apparatus-bottles").style.display = "none";
+    document.getElementById("apparatus-spectrometer").style.display = "none";
+    document.getElementById("animation-video").style.display = "block";
+    document.getElementById("animation-bottom-right").play();
+    document.getElementById("head-instructions").innerHTML = "Observations";
+    document.getElementById("head-observations").innerHTML = "Observations";
+    document.getElementById("observation").innerHTML = "";
+    document.getElementById("instruction").innerHTML = "";
+
+    // Syncing Observation messages with Video Speed
+    let timeOuts = [2000, 5000, 3000, 5000];
+
+    for (let index = 0; index < timeOuts.length; index++) {
+      await new Promise((r) => setTimeout(r, timeOuts[index] * speedFactor));
+      observeMessage();
+    }
+    await new Promise((r) => setTimeout(r, 3000 * speedFactor));
```

Comment:
Is this a candidate for becoming a separate async funtion? This same pattern has been used 3 times in this file.

Reply:
Yes Sir, I had tried to make a different function for syncing the video observations but then the current function was skipping to the below if-block (where it displays to observe the graph) because the current function is not awaiting a promise anymore and the called function will be async so it starts functioning like a normal time delay instead of awaiting promise before executing the rest of the function. 
So I had changed it to a for loop block like you had recommended in the previous code review.